File: js/tests/reverseAndCapitalize.test.js
const reverseCapitalize = require("../reverseAndCapitalize");

const data = [
  ["edabit", "TIBADE"],
  ["abc", "CBA"],
  ["hellothere", "EREHTOLLEH"],
  ["input", "TUPNI"],
  ["indubitably", "YLBATIBUDNI"]
];

describe("reverseCapitalize", () => {
  describe("Return reversed and uppercase string", () => {
    test.each(data)("reverseCapitalize(%s) = %s", (str, result) => {
      let actual = reverseCapitalize(str);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/ReverseAndCapitalizeTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class ReverseAndCapitalizeTest
    {
        [Theory]
        [InlineData("edabit", "TIBADE")]
        [InlineData("abc", "CBA")]
        [InlineData("hellothere", "EREHTOLLEH")]
        [InlineData("input", "TUPNI")]
        [InlineData("indubitably", "YLBATIBUDNI")]
        public void ReverseCapitalize_StringInput_ReturnReversedAndCapitalizedString(string str, string expected)
        {
            var actual = ReverseAndCapitalize.ReverseCapitalize(str);

            Assert.Equal(expected, actual);
        }
    }
}